rs-matter/src/data_model/root_endpoint.rs

```diff
@@ -56,15 +55,83 @@ pub const CLUSTERS: [Cluster<'static>; 10] = [
     group_key_management::CLUSTER,
 ];
 
-pub const fn endpoint(id: EndptId) -> Endpoint<'static> {
+/// The type of operational network (Ethernet, Wifi or (future) Thread)
+/// for which root endpoint meta-data is being requested
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum OperNwType {
+    Ethernet,
+    Wifi,
+}
+
+/// A utility function to create a root (Endpoint 0) object using the requested operational network type.
+pub const fn endpoint(id: EndptId, op_nw_type: OperNwType) -> Endpoint<'static> {
     Endpoint {
         id,
         device_type: super::device_types::DEV_TYPE_ROOT_NODE,
-        clusters: &CLUSTERS,
+        clusters: clusters(op_nw_type),
+    }
+}
+
+/// A utility function to return the clusters for a root (Endpoint 0) object using the requested operational network type.
+pub const fn clusters(op_nw_type: OperNwType) -> &'static [Cluster<'static>] {
+    match op_nw_type {
+        OperNwType::Ethernet => &ETH_NW_CLUSTERS,
+        OperNwType::Wifi => &WIFI_NW_CLUSTERS,
     }
 }
 
-pub fn handler<'a, T>(endpoint_id: u16, matter: &'a T) -> RootEndpointHandler<'a>
+/// A type alias for a root (Endpoint 0) handler using Ethernet as an operational network
+pub type EthRootEndpointHandler<'a> = RootEndpointHandler<'a, EthNwCommCluster, EthNwDiagCluster>;
+
+/// A type representing the type of the root (Endpoint 0) handler
+/// which is generic over the operational transport clusters (i.e. Ethernet, Wifi or Thread)
+pub type RootEndpointHandler<'a, NWCOMM, NWDIAG> = handler_chain_type!(
+    NWCOMM,
+    NWDIAG,
+    HandlerCompat<descriptor::DescriptorCluster<'a>>,
+    HandlerCompat<cluster_basic_information::BasicInfoCluster<'a>>,
+    HandlerCompat<general_commissioning::GenCommCluster<'a>>,
+    HandlerCompat<admin_commissioning::AdminCommCluster<'a>>,
+    HandlerCompat<noc::NocCluster<'a>>,
+    HandlerCompat<access_control::AccessControlCluster<'a>>,
+    HandlerCompat<general_diagnostics::GenDiagCluster>,
+    HandlerCompat<group_key_management::GrpKeyMgmtCluster>
+);
+
+/// A utility function to instantiate the root (Endpoint 0) handler using Ethernet as the operational network.
```

Comment:
similar question here (and it probably repeats throughout): given that this seems specifically for endpoint 0, why do we pass in an endpoint_id. We should either not pass it in or describe why this is done and have the 0 be some form of "typically" documented

Reply:
The only reason is: change less things. The place from where this code is coming already [does have the `endpoint_id` parameter removed](https://github.com/ivmarkov/esp-idf-matter/blob/master/src/stack.rs#L431), but when I was opening it as a PR I reduced the PR surface deliberately.

But if you are sure the root endpoint will always be 0, and there will never be a case where we need something like the root endpoint but with a different ID, I can hard-code to 0.